ce/insights.ts

```diff
@@ -23,18 +25,50 @@ export const insights = setup('b4e88960-4393-4dd9-ab8e-97e8fe6d7603').
   setSendLiveMetrics(false).
   setUseDiskRetryCaching(false).
   start();
-export function trackEvent(name: string, properties?: { [key: string]: string }) {
+
+// todo: This will be refactored to allow appInsights to be called out-of-proc from the main process.
+//       in order to not potentially slow down or block on activation/etc.
+
+export function flushTelemetry() {
+  session.channels.debug('Ensuring Telemetry data is finished sending.');
+  defaultClient.flush({});
+}
+
+defaultClient.addTelemetryProcessor((envelope, contextObjects) => {
+  if (session.context['printmetrics']) {
+    session.channels.message(`Telemetry Event: \n${JSON.stringify(envelope.data, null, 2)}`);
+  }
+
+  // only actually send telemetry if it's enabled.
+  return session.telemetryEnabled;
+});
+
+export function trackEvent(name: string, properties: { [key: string]: string } = {}) {
+  session.channels.debug(`Triggering Telemetry Event ce.${name}`);
   defaultClient.trackEvent({
-    name: 'Microsoft.ApplicationInsights.Event',
+    name: `ce.${name}`,
+    time: new Date(),
+
     tagOverrides: {
       'ai.device.os': process.platform,
       'ai.device.osVersion': version(),
     },
+
     properties: {
       ...properties,
-      'from': 'vcpkg-ce'
+      'kind': 'vcpkg-ce',
+      'ce.version': Version,
```

Comment:
You can send this in `tagOverrides` using `defaultClient.context.keys.applicationVersion`.

Reply:
ok